src/nodes.js

```diff
@@ -0,0 +1,518 @@
+  let x = parseInt(cx)
+  let y = parseInt(cy)
+
+  if (shape === "circle") {
+    var nodeG = d3.select(canvas)
+      .append("g")
+      .attr("class", "node")
+      .attr("id", node.id)
+      .attr("content", "false")
+      .attr("transform", "matrix(1,0,0,1,"+x+","+y+")");
+    nodeG
+      .append(shape)
+      .attr("class", "node-rep")
+      .style("fill", color)
+      .style("z-index", 1)
+      .attr("r", radius)
+      .attr("cx", 0)
+      .attr("cy", 0)
+      .attr("xmlns", "http://www.w3.org/2000/svg");
+  }
+
+  if (shape === "rect") {
+    var nodeG = d3.select(canvas)
+      .append("g")
+      .attr("class", "node")
+      .attr("id", node.id)
+      .attr("content", "false")
+      .attr("transform", "translate("+x+","+y+")");
+    nodeG
+      .append(shape)
+      .attr("class", "node-rep")
+      .style("fill", color)
+      .style("z-index", 1)
+      .attr("width", width)
+      .attr("height", height)
+      .attr("cx", 0)
+      .attr("cy", 0)
+      .attr("xmlns", "http://www.w3.org/2000/svg");
+  }
+  return nodeG.node();
+}
+*/
+
+function drawLink(link) {
+  let linkSrcNode = document.getElementById(link.sourceId);
+  let linkDestNode = document.getElementById(link.destId);
+  //console.log("source node: " + linkSrcNode + ", dest node: " + linkDestNode);
+  let srcPos = getNodePosition(linkSrcNode);
+  let destPos = getNodePosition(linkDestNode);
+
+  //Inserts the link before all the nodes in the canvas
+  var linkG = d3.select(canvas)
+    .insert("g", ".node")
+    .attr("class", "link")
+    .attr("id", link.id)
+    .attr("source_id", link.sourceId)
+    .attr("target_id", link.destId)
+    .attr("content", "false");
+  linkG
+    .append("line")
+    .attr("class", "link-rep")
+    .attr("id", link.id)
+    .attr("x1", srcPos.x)
+    .attr("y1", srcPos.y)
+    .attr("x2", destPos.x)
+    .attr("y2", destPos.y)
+    .attr("xmlns", "http://www.w3.org/2000/svg");
+
+    return linkG.node();
+}
+
+function removeNode(node) {
+  //console.log(node);
+  let node_d3 = node instanceof d3.selection ?  node : d3.select(node);
+  let node_id = node_d3.attr("id");
+
+  d3.selectAll(`[source_id=${node_id}]`)
+    .classed("deleted", true);
+
+  d3.selectAll(`[target_id=${node_id}]`)
+    .classed("deleted", true);
+
+  d3.selectAll(".selection_area[children_ids~=" + node_id + "]")
+    .each(function(){
+      let group = d3.select(this);
+      group.attr("children_ids", group.attr("children_ids").split(' ').filter(id => id !== node_id).join(' ') );
+    });
+
+  closePreviewIframe("node");
+  node_d3.classed("deleted", true);
+  let temp_transient = document.createElement("transient");
+  temp_transient.appendChild(node_d3.node());
+  document.getElementById("canvas").appendChild(temp_transient);
+}
+
+function removeLink(link) {
+  link = link instanceof d3.selection ? link : d3.select(link);
+  let link_id = link.attr("id");
+  closePreviewIframe("edge");
```

Comment:
No longer need closePreviewIframe since the notes functionality is in container now.